sql/core/src/test/scala/org/apache/spark/sql/DatasetSuite.scala

```diff
@@ -2909,6 +3016,7 @@ object KryoData {
 
 /** Used to test Java encoder. */
 class JavaData(val a: Int) extends Serializable {
+  def this() = this(0)
```

Comment:
It is not clear to me my why the method 'def this()' needs to be explicitly defnined. I don't see it being used anywhere. Can we remove it or Could you please explain why this is here?

Reply:
Any class which is Serializable, for serializer to work correctly, needs to have a default empty argument constructor.
Otherwise there will be runtime exception.